BBj-archetype engine issue

When using the archetype command to get a version of the bbj archetype.

 mvn -B archetype:generate `
>> -DarchetypeGroupId="com.webforj" `
>> -DarchetypeArtifactId="webforj-archetype-bbj-hello-world" `
>> -DarchetypeVersion="LATEST" `
>> -DgroupId="org.example" `
>> -DartifactId="bbjDemo" `
>> -Dversion="1.0-SNAPSHOT" `
>> -DappName="MyApp"

Afterwards running mvn install would succeed, but trying to acces the app leads to the following exception.

java.lang.NoClassDefFoundError: com/webforj/servlet/WebforjServlet
    at com.webforj.Environment.getInitialConfig(Environment.java:637)
    at com.webforj.Environment.init(Environment.java:91)
    at com.webforj.Bootstrap.init(Bootstrap.java:70)
    at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
    at java.base/java.lang.reflect.Method.invoke(Method.java:580)
    at com.basis.startup.LoadLibrary.invokeMethodInContext(LoadLibrary.java:23)
    at com.basis.util.common.security.l$a.invokeMethod(l$a.java:17)
    at com.basis.bbj.processor.program.type.JavaType$JavaMethod.invoke(JavaType.java:4)
    at com.basis.bbj.processor.program.type.SignatureHandler.dispatch(SignatureHandler.java:119)
    at com.basis.bbj.processor.program.type.JavaType$JavaSignature.dispatch(JavaType.java:1)
    at com.basis.bbj.processor.program.type.SignatureHandler.handleSignature(SignatureHandler.java:118)
    at com.basis.bbj.processor.instruction.a.k.a(k.java:88)
    at com.basis.bbj.processor.instruction.a.k.a(k.java:120)
    at com.basis.bbj.processor.instruction.ee.a(ee.java:40)
    at com.basis.bbj.processor.instruction.c.bw.k(bw.java:2498)
    at com.basis.bbj.processor.instruction.c.bw.i(bw.java:1790)
    at com.basis.bbj.processor.instruction.c.bw.run(bw.java:2628)
    at com.basis.util.common.BasisThread.run(BasisThread.java:119)
Caused by: java.lang.ClassNotFoundException: com.webforj.servlet.WebforjServlet
    at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
    at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:593)
    at com.basis.util.common.security.SessionClassLoader.loadClass(SessionClassLoader.java:360)
    at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
    ... 18 more

Removing the exclusion of the engine in the pom solves that issue, but then running the app leads to this exception:

om.webforj.router.exception.NotFoundException: Failed to match route for location: /webapp/bbjtest
    at com.webforj.router.Router.navigate(Router.java:183)
    at com.webforj.router.Router.navigate(Router.java:283)
    at com.webforj.App.lambda$resolveFirstRoute$4(App.java:782)
    at java.base/java.util.Optional.ifPresent(Optional.java:178)
    at com.webforj.App.resolveFirstRoute(App.java:779)
    at com.webforj.App.initialize(App.java:202)
    at com.webforj.Bootstrap.initApplication(Bootstrap.java:110)
    at com.webforj.Bootstrap.init(Bootstrap.java:78)
    at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
    at java.base/java.lang.reflect.Method.invoke(Method.java:580)
    at com.basis.startup.LoadLibrary.invokeMethodInContext(LoadLibrary.java:23)
    at com.basis.util.common.security.l$a.invokeMethod(l$a.java:17)
    at com.basis.bbj.processor.program.type.JavaType$JavaMethod.invoke(JavaType.java:4)
    at com.basis.bbj.processor.program.type.SignatureHandler.dispatch(SignatureHandler.java:119)
    at com.basis.bbj.processor.program.type.JavaType$JavaSignature.dispatch(JavaType.java:1)
    at com.basis.bbj.processor.program.type.SignatureHandler.handleSignature(SignatureHandler.java:118)
    at com.basis.bbj.processor.instruction.a.k.a(k.java:88)
    at com.basis.bbj.processor.instruction.a.k.a(k.java:120)
    at com.basis.bbj.processor.instruction.ee.a(ee.java:40)
    at com.basis.bbj.processor.instruction.c.bw.k(bw.java:2498)
    at com.basis.bbj.processor.instruction.c.bw.i(bw.java:1790)
    at com.basis.bbj.processor.instruction.c.bw.run(bw.java:2628)
    at com.basis.util.common.BasisThread.run(BasisThread.java:119)

--- a/webforj-foundation/src/main/java/com/webforj/Environment.java
+++ b/webforj-foundation/src/main/java/com/webforj/Environment.java
@@ -632,11 +632,17 @@ public final class Environment {
           ConfigFactory.parseFile(configPath.toFile()).withFallback(getInitialDefaultConfig());
     }
 
-    // Merge with config from WebforjServlet if available (e.g., from Spring Boot)
+    // Merge with config from WebforjServlet if available
     // The servlet config takes precedence over file-based config
-    Config servletConfig = WebforjServlet.getInitConfig();
-    if (servletConfig != null) {
-      theConfig = servletConfig.withFallback(theConfig);
+    try {
+      Config servletConfig = WebforjServlet.getInitConfig();
+      if (servletConfig != null) {
+        theConfig = servletConfig.withFallback(theConfig);
+      }
+    } catch (NoClassDefFoundError e) {
+      // WebforjServlet is not available in this environment (no servlet container)
+      logger.log(Level.DEBUG, "WebforjServlet not available, skipping servlet config: {0}",
+          e.getMessage());
     }
 
     return theConfig;

